<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Function reference â€¢ scanalysis</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Function reference" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body>
    <div class="container template-reference-index">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">scanalysis</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/preprocessing.html">Preprocessing Visualizations</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/keshavmot2/scanalysis">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index">

    <colgroup>
      
      <col class="alias" />
      <col class="title" />
    </colgroup>

    <tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-all-functions" class="hasAnchor"><a href="#section-all-functions" class="anchor"></a>All functions</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      <tr>
        
        <td>
          <p><code><a href="annotate_cdr3.html">annotate_cdr3()</a></code> </p>
        </td>
        <td><p>Annotate CDR3 sequences stored in colData(sce)$vdj based on a reference dataset (for example, Ag-specific data)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="annotate_chain_count.html">annotate_chain_count()</a></code> </p>
        </td>
        <td><p>Annotate number of TRA, TRB, IGL, and IGH reads per barcode</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="annotate_n_genes_expr.html">annotate_n_genes_expr()</a></code> </p>
        </td>
        <td><p>Add number of genes expressed per barcode to colData(sce)$n_genes_expr</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="annotate_num_cells_expr.html">annotate_num_cells_expr()</a></code> </p>
        </td>
        <td><p>Add number of cells with nonzero expression for each gene to rowData(sce)$num_cells_expr</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="annotate_pct_gene_set.html">annotate_pct_gene_set()</a></code> </p>
        </td>
        <td><p>Add percent of gene set defined by a regular expression pattern to colData(sce)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="annotate_pct_total_reads.html">annotate_pct_total_reads()</a></code> </p>
        </td>
        <td><p>Add percentage of total reads that each gene takes up across the whole dataset to rowData(sce)$pct_reads</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="annotate_total_umi_count.html">annotate_total_umi_count()</a></code> </p>
        </td>
        <td><p>Add number of total umis per barcode to colData(sce)$total_umi</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="assign_clonotypes.html">assign_clonotypes()</a></code> </p>
        </td>
        <td><p>Function to use for assigning clonotypes based on custom definition</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="compute_evenness_profile_long.html">compute_evenness_profile_long()</a></code> </p>
        </td>
        <td><p>Compute tidied data frame of evenness profiles per group based on clonotype frequency distributions</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="compute_evenness_profile_matrix.html">compute_evenness_profile_matrix()</a></code> </p>
        </td>
        <td><p>Compute matrix of evenness profiles per group based on clonotype frequency distributions</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="convert_identity_frequency_matrix_to_long.html">convert_identity_frequency_matrix_to_long()</a></code> </p>
        </td>
        <td><p>Convert an identity frequency matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dot-.get_cell_features_coldata.html">..get_cell_features_coldata()</a></code> </p>
        </td>
        <td><p>Get feature from column metadata</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dot-encode_identity_frequency_matrix.html">.encode_identity_frequency_matrix()</a></code> </p>
        </td>
        <td><p>Helper function to encode identity frequency matrix given data frame</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dot-filter_ambient_barcode_cellranger.html">.filter_ambient_barcode_cellranger()</a></code> </p>
        </td>
        <td><p>Get filter for cells based on inflection point</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dot-filter_ambient_barcode_empty_drops.html">.filter_ambient_barcode_empty_drops()</a></code> </p>
        </td>
        <td><p>Get filter for cells based on emptyDrops algorithm</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dot-filter_ambient_barcode_inflection.html">.filter_ambient_barcode_inflection()</a></code> </p>
        </td>
        <td><p>Get filter for cells based on inflection point</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dot-filter_ambient_barcode_knee.html">.filter_ambient_barcode_knee()</a></code> </p>
        </td>
        <td><p>Get filter for cells based on knee point</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dot-get_cell_features_assay.html">.get_cell_features_assay()</a></code> </p>
        </td>
        <td><p>Get feature from assay (from alternate experiment)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dot-get_cell_features_reduced_dimensions.html">.get_cell_features_reduced_dimensions()</a></code> </p>
        </td>
        <td><p>Get feature from reduced dimensional representation</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dot-get_reduced_dims.html">.get_reduced_dims()</a></code> </p>
        </td>
        <td><p>Get reduced dimensions of object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dot-prepare_barcode_counts.html">.prepare_barcode_counts()</a></code> </p>
        </td>
        <td><p>Prepare count annotations for number of cells passing each filtering method</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dot-prepare_barcode_cutoffs.html">.prepare_barcode_cutoffs()</a></code> </p>
        </td>
        <td><p>Prepare cutoff values for each method</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dot-prepare_barcode_plot_data.html">.prepare_barcode_plot_data()</a></code> </p>
        </td>
        <td><p>Prepare data for plotting barcode QC plot</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dot-prepare_gex_bivariate_counts.html">.prepare_gex_bivariate_counts()</a></code> </p>
        </td>
        <td><p>Prepare counts of cells that meet filter criteria to annotate based on two filters</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dot-prepare_gex_data.html">.prepare_gex_data()</a></code> </p>
        </td>
        <td><p>Prepare data frame to plot for gene expression related QC plots</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dot-prepare_gex_scale_limits.html">.prepare_gex_scale_limits()</a></code> </p>
        </td>
        <td><p>Prepare the limits of plot scale</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dot-prepare_gex_univariate_counts.html">.prepare_gex_univariate_counts()</a></code> </p>
        </td>
        <td><p>Prepare counts of cells that meet filter criteria to annotate based on a single filter</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dot-prepare_volcano_annotations.html">.prepare_volcano_annotations()</a></code> </p>
        </td>
        <td><p>Prepare annotations for volcano plot</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dot-prepare_volcano_top_annotations.html">.prepare_volcano_top_annotations()</a></code> </p>
        </td>
        <td><p>Prepare the top N annotations for volcano plot</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dot-sce_to_assay.html">.sce_to_assay()</a></code> </p>
        </td>
        <td><p>Convert SingleCellExperiment object to Seurat assay</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dot-seurat_assay_to_sce.html">.seurat_assay_to_sce()</a></code> </p>
        </td>
        <td><p>Convert Seurat assay to SingleCellExperiment</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="encode_cell_identity_frequency_long.html">encode_cell_identity_frequency_long()</a></code> </p>
        </td>
        <td><p>Encode frequencies of (combinations) of values in columns from colData into a long data frame</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="encode_cell_identity_frequency_matrix.html">encode_cell_identity_frequency_matrix()</a></code> </p>
        </td>
        <td><p>Encode frequencies of (combinations) of values in columns from colData into a matrix with rows as groups and columns as features</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="encode_vdj_identity_frequency_long.html">encode_vdj_identity_frequency_long()</a></code> </p>
        </td>
        <td><p>Encode frequencies of (combinations) of values in columns from colData into long data frame</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="encode_vdj_identity_frequency_matrix.html">encode_vdj_identity_frequency_matrix()</a></code> </p>
        </td>
        <td><p>Encode frequencies of (combinations) of values in columns from colData into a matrix with rows as groups and columns as features</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="filter_ambient_barcode.html">filter_ambient_barcode()</a></code> </p>
        </td>
        <td><p>Get filter to remove ambient RNA barcodes based on the combination of multiple methods:</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="filter_chain_count.html">filter_chain_count()</a></code> </p>
        </td>
        <td><p>Get filter for barcodes based on the number of chains for TRA, TRB, IGL, and IGH present</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="filter_n_genes_expr.html">filter_n_genes_expr()</a></code> </p>
        </td>
        <td><p>Get filter for number of genes expressed per barcode based on log transformed values</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="filter_pct_mito.html">filter_pct_mito()</a></code> </p>
        </td>
        <td><p>Get filter for percentage of mitochondrial reads expressed per barcode</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="filter_total_umi.html">filter_total_umi()</a></code> </p>
        </td>
        <td><p>Get filter for total umis per barcode based on log transformed values outside (either above, below, or both based on type parameter) nmads median absolute deviations from the median</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_assay_data.html">get_assay_data()</a></code> </p>
        </td>
        <td><p>Get assay data from either the main experiment or altExps</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_cell_features.html">get_cell_features()</a></code> </p>
        </td>
        <td><p>Get feature from assay data, colData, or reducedDims at once from main experiment or alternate experiments</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="hello.html">hello()</a></code> </p>
        </td>
        <td><p>Hello, World!</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_barcode_qc.html">plot_barcode_qc()</a></code> </p>
        </td>
        <td><p>QC plot of UMI rank vs total number of UMIs per barcode</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_distributions.html">plot_distributions()</a></code> </p>
        </td>
        <td><p>Plot feature distributions from SingleCellExperiment objects</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_pairwise_scatterplot.html">plot_pairwise_scatterplot()</a></code> </p>
        </td>
        <td><p>Plot pairwise scatterplot of cell-level data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_qc_distribution.html">plot_qc_distribution()</a></code> </p>
        </td>
        <td><p>Scatterplot of two features of interest from colData with annotated thresholds and counts based on filters</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_qc_scatterplot.html">plot_qc_scatterplot()</a></code> </p>
        </td>
        <td><p>Scatterplot of two features of interest from colData with annotated thresholds and counts based on filters</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_reduced_dimensions.html">plot_reduced_dimensions()</a></code> </p>
        </td>
        <td><p>Plot reduced dimensional plot with multiple features</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_vdj_qc.html">plot_vdj_qc()</a></code> </p>
        </td>
        <td><p>Plot number of VDJ chains versus colData variable of interest with annotated filters</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_volcano.html">plot_volcano()</a></code> </p>
        </td>
        <td><p>Plot volcano plot with annotations</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="read_10x.html">read_10x()</a></code> </p>
        </td>
        <td><p>Read 10X gene expression and VDJ data into a SingleCellExperiment object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="read_10x_gene_expr.html">read_10x_gene_expr()</a></code> </p>
        </td>
        <td><p>Read gene expression matrix from 10X data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="read_10x_vdj.html">read_10x_vdj()</a></code> </p>
        </td>
        <td><p>Read VDJ information from 10X data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sce_to_seurat.html">sce_to_seurat()</a></code> </p>
        </td>
        <td><p>Convert SingleCellExperiment object to Seurat and retain multi-modal data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="seurat_to_sce.html">seurat_to_sce()</a></code> </p>
        </td>
        <td><p>Convert Seurat object to SingleCellExperiment and retain multi-modal data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="unnest_vdj.html">unnest_vdj()</a></code> </p>
        </td>
        <td><p>Get VDJ data with cell-level metadata</p></td>
      </tr>
    </tbody>
    </table>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#section-all-functions">All functions</a></li>
    </ul>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Keshav Motwani <keshav.motwani@pathology.ufl.edu>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


